test/org/sagebionetworks/bridge/stormpath/StormpathDirectoryDaoTest.java

```diff
@@ -127,38 +119,24 @@ private void assertDirectoriesAreEqual(DynamoStudy study, String rpSubject, Stri
-        CloseableHttpClient client = HttpClientBuilder.create().setDefaultCredentialsProvider(provider).build();
-        
-        // Get directory as JSON
-        ObjectNode directoryNode = StormpathUtils.getJSON(client, directory.getHref());
-        String accountCreationUrl = directoryNode.get("accountCreationPolicy").get("href").asText();
-        
-        // Get account policy as JSON, update to our standard configuration
-        ObjectNode accountPolicyNode = StormpathUtils.getJSON(client, accountCreationUrl);
-        String verificationEmailTemplatesUrl = accountPolicyNode.get("verificationEmailTemplates").get("href").asText();
+        AccountCreationPolicy policy = newDirectory.getAccountCreationPolicy();
+        assertEquals(EmailStatus.ENABLED, policy.getVerificationEmailStatus());
+        assertEquals(EmailStatus.DISABLED, policy.getWelcomeEmailStatus());
+        assertEquals(1, policy.getAccountVerificationEmailTemplates().getSize());
         
-        // Get the verify email template
-        ObjectNode templateNode = StormpathUtils.getJSON(client, verificationEmailTemplatesUrl);
-        String templateUrl = templateNode.get("items").get(0).get("href").asText();
-        // Update this template with study-specific information
-        ObjectNode templateJSON = StormpathUtils.getJSON(client, templateUrl);
-        
-        // TODO: Test that the study name is used if no study name is provided.
-        assertEquals(study.getSponsorName(), templateJSON.get("fromName").asText());
-        assertEquals(study.getSupportEmail(), templateJSON.get("fromEmailAddress").asText());
-        assertEquals(ve.getSubject(), templateJSON.get("subject").asText());
-        assertEquals(ve.getMimeType().toString(), templateJSON.get("mimeType").asText());
-        assertEquals(ve.getBody(), templateJSON.get("textBody").asText());
-        
-        assertTrue(templateJSON.get("defaultModel").get("linkBaseUrl").asText().contains("/mobile/verifyEmail.html?study="));
+        template = policy.getAccountVerificationEmailTemplates().iterator().next();
+        assertEquals(study.getSponsorName(), template.getFromName());
+        assertEquals(study.getSupportEmail(), template.getFromEmailAddress());
+        assertEquals(veSubject, template.getSubject());
+        assertEquals(getStormpathMimeType(study.getVerifyEmailTemplate()), template.getMimeType());
+        assertEquals(study.getVerifyEmailTemplate().getBody(), template.getTextBody());
+        url = String.format("%s/mobile/verifyEmail.html?study=%s", BridgeConfigFactory.getConfig().getBaseURL(), study.getIdentifier());
+        assertEquals(url, template.getLinkBaseUrl());
+    }
+    
+    private com.stormpath.sdk.mail.MimeType getStormpathMimeType(EmailTemplate template) {
+        return (template.getMimeType() == EmailTemplate.MimeType.TEXT) ? 
+            com.stormpath.sdk.mail.MimeType.PLAIN_TEXT : com.stormpath.sdk.mail.MimeType.HTML;
```

Comment:
This method is identical to the one in StormpathDirectoryDao. Should we make it non-private so it can be used by tests?